packages/components/src/shared/src/dom.ts

```diff
@@ -0,0 +1,9 @@
+export function canUseDOM(): boolean {
```

Comment:
I honestyle hate this code lol. Not a big fan of `!!`, should `isNil` to be consistant with the codebase.